### PR TITLE
Update to Protocol 12.0.0 Fix #9075 

### LINKED ISSUE #9075: Update to Protocol 12.0.0
Update to Protocol 12.0.0.

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description

Update to protocol 12.0.0. This was a [major release](https://github.com/mozilla/protocol/releases/tag/v12.0.0) and impacted all pages with forms on them.

## Issue / Bugzilla link

Fix #9075 

## Testing

There's 15 commit here, reviewing them separately will make it easier to understand what was changed and why. Git inserts handy links to do this below this box of prose.

Please pay special attention to error and success states since those are not always obvious when looking at the page.

### Protocol 12.0.0.

- yarn and package lock files updated
- bedrock compiles

### Adjust to new content widths

- find and replace `mzp-t-narrow` to `mzp-t-content-md`
- deal with one case of mzp-l-narrow
- reduced width of:
    - https://www-demo4.allizom.org/en-US/newsletter/recovery/
    - https://www-demo4.allizom.org/en-US/firefox/ios/testflight/

### Adjust to new button sizes

- find an replace `mzp-t-small` to `mzp-t-md` on buttons
- enlarged some default sized buttons after visual survey

### Change all occurances of errorlist to mzp-c-form-error

- in some places python inserts the class automatically, those are unchanged

### Rely on new form styles for bottom margin under inputs

- changed paragraphs to divs where appropriate

### Update FxA form styles to use form component

https://www-demo4.allizom.org/en-US/firefox/accounts/
https://www-demo4.allizom.org/en-US/firefox/browsers/
https://www-demo4.allizom.org/en-US/firefox/products/

### Update send to device form styles

https://www-demo4.allizom.org/en-US/firefox/mobile/ (modal)
https://www-demo4.allizom.org/en-US/firefox/mobile/get-app/
https://www-demo4.allizom.org/en-US/firefox/72.0.0beta/whatsnew/all/
https://www-demo4.allizom.org/en-US/firefox/73.0.0/whatsnew/all/ (modal)
https://www-demo4.allizom.org/en-US/firefox/welcome/6/ (modal)
https://www-demo4.allizom.org/en-US/firefox/mobile-promo/ (modal)

### Update newsletter signup form styles

https://www-demo4.allizom.org/en-US/firefox/developer/
https://www-demo4.allizom.org/en-US/firefox/78.0.0a2/whatsnew/all/
https://www-demo4.allizom.org/en-US/firefox/retention/thank-you/
https://www-demo4.allizom.org/en-US/foundation/trademarks/policy/
https://www-demo4.allizom.org/en-US/about/legal/
https://www-demo4.allizom.org/en-US/contribute/
https://www-demo4.allizom.org/en-US/firefox/channel/desktop/

### Update newsletter settings form styles

https://www-demo4.allizom.org/en-US/newsletter/developer/
https://www-demo4.allizom.org/en-US/newsletter/firefox/
https://www-demo4.allizom.org/en-US/newsletter/mozilla/
https://www-demo4.allizom.org/en-US/newsletter/opt-out-confirmation/
https://www-demo4.allizom.org/en-US/newsletter/recovery/
https://www-demo4.allizom.org/en-US/newsletter/updated/

### Update first run page form styles

- I had to make some adjustments for the new content widths
- https://www-demo4.allizom.org/en-US/firefox/firstrun/

### Update fraud report form styles

https://www-demo4.allizom.org/en-US/about/legal/fraud-report/

### Update speaker request form styles

https://www-demo4.allizom.org/en-US/press/speakerrequest/

### Update Enterprise signup form styles

https://www-demo4.allizom.org/en-US/firefox/enterprise/signup/
https://www-demo4.allizom.org/en-US/firefox/enterprise/signup/thanks/

### CODE REVIEW

**Comment by alexgibson** [comment]
I took a first visual pass before looking into code. Some of these may be intentional, others are certainly not blockers. Just calling them out incase something is incorrect (it's pretty hard to tell in some cases). I'll add things in comments below as I find them.

**Comment by alexgibson** [comment]
The download buttons in the nav have a new size, so they are no longer vertically centered:

image

**Comment by alexgibson** [comment]
The max-width of `mzp-l-content` has changed from `1420px` to `1440px`, which in itself is fine. But the max-width of the navigation has not changed to match?

image

**Comment by alexgibson** [comment]
There's still one occurence of `mzp-t-small` here: https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx71.html#L41

**Comment by alexgibson** [comment]
Color contrast issue on send-to-device success message:

image

http://localhost:8000/en-US/firefox/50.0/whatsnew/all/

File: media/css/firefox/channel.scss
Comment (alexgibson) [inline media/css/firefox/channel.scss:23]: This looks / feels pretty squished? `$content-lg` seems closer to what's in production.

**Comment by alexgibson** [comment]
Secondary navigation is no longer aligned with the primary navigation:

image

File: bedrock/firefox/templates/firefox/enterprise/index.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/enterprise/index.html:51]: Curious why we add `mxp-t-xl` here, but pages like `/firefox/new/` still have the default (smaller) button sizes? Is there a new rule that should be followed here?

image


image

**Comment by alexgibson** [comment]
The `mzp-l-card-half` layout on the home page has some alignment issues with the new content max-width:

image

**Comment by alexgibson** [comment]
Some text replacement issues on the footer social icons:

image

File: media/css/firefox/browsers-products.scss
Comment (alexgibson) [inline media/css/firefox/browsers-products.scss:597]: Curious, does this go further by addressing something that some extra `padding-top` couldn't fix?

**Comment by stephaniehobson** [comment]
Blocked by https://github.com/mozilla/protocol/issues/611

File: media/css/firefox/browsers-products.scss
Comment (stephaniehobson) [reply media/css/firefox/browsers-products.scss:597]: It will reflect the text size, including the user's personal settings, more accurately than a px or em padding value.

File: bedrock/firefox/templates/firefox/enterprise/index.html
Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/enterprise/index.html:51]: No rule, I was just going with my gut.

File: media/css/firefox/channel.scss
Comment (stephaniehobson) [reply media/css/firefox/channel.scss:23]: There wasn't a really good fit for this one. I agree lg is closer to what's in production but production already has a line length that is too wide for readability.

I would have liked to make the call outs their default width and keep the supplementary info narrower but the text inside the callout was not inline with the other info. I'll look into re-structuring the page.

Comment (stephaniehobson) [reply media/css/firefox/channel.scss]: Changed it to $content-lg

**Comment by stephaniehobson** [comment]
This is ready for re-review.

File: bedrock/firefox/templates/firefox/enterprise/index.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/enterprise/index.html:51]: Ok, well if there's one button where I think people might freak out a little if we made it smaller, it's probably the one on /new. Worth making it match the size you've applied to /enterprise?

**Comment by alexgibson** [comment]
The copy inside the nav download button feels a little off center? (screenshot from Chrome)

image

**Comment by alexgibson** [comment]
Oh, suggest we also push this to run-integration-tests prior to merging?

r+wc

**Comment by stephaniehobson** [comment]
The nav button looks off center because there are no descenders in the words "Download Firefox". Normally I wouldn't recommend a fix for specific wording in a specific language but since this button is on every page I could make an exception. I'll talk to @dzingeek about it.

**Comment by stephaniehobson** [comment]
🟢 Passing test run: https://gitlab.com/mozmeao/www-config/-/pipelines/173173828

**Comment by stephaniehobson** [comment]
I changed the download macro to make the download button xl by default. This touched a number of pages but feels like a better approach.

**Comment by alexgibson** [comment]
Needs a rebase after https://github.com/mozilla/bedrock/pull/9169 merged

**Comment by stephaniehobson** [comment]
Rebased.

**Comment by alexgibson** [comment]
Ugh, sorry look like this needs rebasing again. Let's try and get this one in before another large thing merges.

**Comment by stephaniehobson** [comment]
Do not merge. Going through looking for new content that was added after the conversion PR started.

**Comment by stephaniehobson** [comment]
I'm going to leave this until I'm back. @craigcook thought he might pick it up so here is what needs to be done:

These were done one already but it's been a month so it's just reviewing new content"

- [ ] `button_color` -> `button_class`
- [ ] `mzp-t-small` -> `mzp-t-md`
- [ ] `*-narrow` -> use new content size widths
- [ ] vertically center text in navigation menu download button on EN pages (optional, and I only suggest making this change because it's everywhere and obvious)
- [ ] look at newsletter stuff to make sure conflicts with #9172 were resolved correctly (I'd just compare the diffs in github)
- [ ] why was there a conflict with the fraud report, was that resolved correctly?
- [ ] look at commits made in the last month to make sure new content is displaying correctly

**Comment by stephaniehobson** [comment]
R+ to Craig's amendments.

**Review by alexgibson — APPROVED** [review]

**Pull request merged**
